api_tests/nodes/views/test_node_sparse_fieldsets.py

```diff
@@ -1,76 +1,88 @@
 from osf_tests.factories import (
     ProjectFactory,
     AuthUserFactory,
+    PrivateLinkFactory,
 )
-from api_tests.nodes.views.test_view_only_query_parameter import ViewOnlyTestCase
 
+@pytest.fixture()
+def user():
+    return AuthUserFactory()
 
-class TestNodeSparseFieldsList(ApiTestCase):
-    def setUp(self):
-        super(TestNodeSparseFieldsList, self).setUp()
-        self.user = AuthUserFactory()
+@pytest.mark.django_db
+class TestNodeSparseFieldsList:
 
-        self.non_contrib = AuthUserFactory()
+    @pytest.fixture()
+    def deleted_project(self):
+        return ProjectFactory(is_deleted=True)
 
-        self.deleted = ProjectFactory(is_deleted=True)
-        self.private = ProjectFactory(is_public=False, creator=self.user)
-        self.public = ProjectFactory(is_public=True, creator=self.user)
+    @pytest.fixture()
+    def private_project(self, user):
+        return ProjectFactory(is_public=False, creator=user)
 
-        self.url = '/{}nodes/?fields[nodes]='.format(API_BASE)
+    @pytest.fixture()
+    def public_project(self, user):
+        return ProjectFactory(is_public=True, creator=user)
 
-    def test_empty_fields_returns_no_attributes(self):
-        res = self.app.get(self.url)
+    @pytest.fixture()
+    def url(self):
+        return '/{}nodes/?fields[nodes]='.format(API_BASE)
+
+    def test_node_sparse_fields_list(self, app, user, deleted_project, private_project, public_project, url):
+
+    #   test_empty_fields_returns_no_attributes
+        res = app.get(url)
         node_json = res.json['data'][0]
 
         assert node_json['attributes'] == {}
         assert set(node_json.keys()) == set(['links', 'type', 'id', 'attributes'])
 
-    def test_sparse_fields_includes_relationships(self):
-        res = self.app.get(self.url + 'children')
+    #   test_sparse_fields_includes_relationships
+        res = app.get(url + 'children')
         node_json = res.json['data'][0]
 
         assert node_json['attributes'] == {}
         assert set(node_json.keys()) == set(['links', 'type', 'id', 'attributes', 'relationships'])
-        assert node_json['relationships']['children']['links']['related']['href'].endswith('/{}nodes/{}/children/'.format(API_BASE, self.public._id))
+        assert node_json['relationships']['children']['links']['related']['href'].endswith('/{}nodes/{}/children/'.format(API_BASE, public_project._id))
 
-    def test_returns_expected_nodes(self):
-        res = self.app.get(self.url + 'title')
+    #   test_returns_expected_nodes
+        res = app.get(url + 'title')
         node_json = res.json['data']
 
         ids = [each['id'] for each in node_json]
-        assert self.public._id in ids
-        assert self.deleted._id not in ids
-        assert self.private._id not in ids
+        assert public_project._id in ids
+        assert deleted_project._id not in ids
+        assert private_project._id not in ids
 
         assert len(node_json) == 1
         node_json = node_json[0]
```

Comment:
rename the first node_json to `resp_json` or something, this reassignment is unpleasant.